pylint/plugins/hass_enforce_type_hints.py

```diff
@@ -398,64 +403,69 @@ class ClassTypeHintMatch:
                     },
-                    return_type="FlowResult",
+                    return_type=SentinelValues.UNDEFINED,  # already covered by async_step_*
                 ),
                 TypeHintMatch(
                     function_name="async_step_hassio",
                     arg_types={
                         1: "HassioServiceInfo",
                     },
-                    return_type="FlowResult",
+                    return_type=SentinelValues.UNDEFINED,  # already covered by async_step_*
                 ),
                 TypeHintMatch(
                     function_name="async_step_homekit",
                     arg_types={
                         1: "ZeroconfServiceInfo",
                     },
-                    return_type="FlowResult",
+                    return_type=SentinelValues.UNDEFINED,  # already covered by async_step_*
                 ),
                 TypeHintMatch(
                     function_name="async_step_mqtt",
                     arg_types={
                         1: "MqttServiceInfo",
                     },
-                    return_type="FlowResult",
+                    return_type=SentinelValues.UNDEFINED,  # already covered by async_step_*
                 ),
                 TypeHintMatch(
                     function_name="async_step_reauth",
                     arg_types={
                         1: "Mapping[str, Any]",
                     },
-                    return_type="FlowResult",
+                    return_type=SentinelValues.UNDEFINED,  # already covered by async_step_*
                 ),
                 TypeHintMatch(
                     function_name="async_step_ssdp",
                     arg_types={
                         1: "SsdpServiceInfo",
                     },
-                    return_type="FlowResult",
+                    return_type=SentinelValues.UNDEFINED,  # already covered by async_step_*
                 ),
                 TypeHintMatch(
                     function_name="async_step_usb",
                     arg_types={
                         1: "UsbServiceInfo",
                     },
-                    return_type="FlowResult",
+                    return_type=SentinelValues.UNDEFINED,  # already covered by async_step_*
                 ),
                 TypeHintMatch(
                     function_name="async_step_zeroconf",
                     arg_types={
                         1: "ZeroconfServiceInfo",
                     },
-                    return_type="FlowResult",
+                    return_type=SentinelValues.UNDEFINED,  # already covered by async_step_*
                 ),
             ],
         ),
     ],
 }
 # Overriding properties and functions are normally checked by mypy, and will only
 # be checked by pylint when --ignore-missing-annotations is False
+_BASE_DEVICE_CLASS = TypeHintMatch(
+    function_name="device_class",
+    return_type=["str", None],
+)
```

Comment:
With my approach, it should be possible to move this back into `_ENTITY_MATCH`.

Reply:
Done